doc/docs.md

```diff
@@ -5719,6 +5719,27 @@ x.v:4:5: error: Linux is not supported
     6 |
 ```
 
+### Compile time types
+
+Compile time types group multiple types into a general higher-level type. This is useful in
+functions with generic parameters, where the input type must have a specific property, for example
+the `.len` attribute in arrays.
+
+V supports the following compile time types:
+
+- `$map` => matches [Maps](#maps).
+- `$int` => matches `int`, `i8`, `i16`, `i32`, `i64`, `u8`, `u16`, `u32`, `u64`, `isize`, `usize`
+  and integer literals.
+- `$float` => matches `f32`, `f64` and float literals.
+- `$struct` => matches [Structs](#structs).
+- `$interface` => matches [Interfaces](#interfaces).
+- `$array` => matches [Arrays](#arrays) and [Fixed Size Arrays](#fixed-size-arrays).
+- `$sumtype` => matches [Sum Types](#sum-types).
+- `$enum` => matches [Enums](#enums).
+- `$alias` => matches [Type aliases](#type-aliases).
+- `$function` => matches [Function Types](#function-types).
+- `$option` => matches [Option Types](#optionresult-types-and-error-handling).
+
```

Comment:
No `$string`?  Also, please alphabetize the list.  Easier to read for most humans, especially if they're searching for something in particular.  :-)

Reply:
`$string` doesn't exist, probably because it would just be matching one type and I'll alphabetize them, yeah